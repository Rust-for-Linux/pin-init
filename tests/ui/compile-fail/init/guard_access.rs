use pinned_init::*;

struct Foo {
    x: usize,
    y: usize,
}

fn main() {
    let _ = init!(Foo {
        x: 0,
        y: {
            let _ = __x_guard;
            0
        },
    });
}
