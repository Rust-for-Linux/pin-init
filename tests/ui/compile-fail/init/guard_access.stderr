error[E0425]: cannot find value `__x_guard` in this scope
  --> tests/ui/compile-fail/init/guard_access.rs:12:21
   |
12 |             let _ = __x_guard;
   |                     ^^^^^^^^^ not found in this scope
